health_check/contrib/http_ping/backends.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import requests
+
+from requests.exceptions import Timeout
+
+from django.conf import settings
+
+from health_check.backends import BaseHealthCheckBackend
+from health_check.exceptions import ServiceUnavailable
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPPingHealthCheck(BaseHealthCheckBackend):
+    """Health check using http requests """
+
+
+    def check_status(self):
+        """Check http conections for all urls in settings.HTTP_ENDPOINTS_HEALTH_CHECK"""
+
+        http_endpoints = getattr(settings, "HTTP_ENDPOINTS_HEALTH_CHECK", [])
+        timeout = getattr(settings, "HTTP_TIMEOUT_HEALTH_CHECK", 1)
+
+        logger.debug("Got %s as the http_endpoints. ", http_endpoints)
+
+        for http_endpoint in http_endpoints:
```

Comment:
what are your opinions on using multiple threads for each endpoint, somthing like:

```
    def check_status(self):
        errors = []

        def _check_status(http_endpoint):
            return requests.get(http_endpoint, timeout=timeout)

        with ThreadPoolExecutor(max_workers=len(self.http_endpoints) or 1) as executor:
            for http_endpoint in executor.map(_check_status, self.http_endpoints):
                if self.critical_service:
                    if not HEALTH_CHECK["WARNINGS_AS_ERRORS"]:
                        self.errors.extend(
                            e
                            for e in plugin.errors
                            if not isinstance(e, ServiceWarning)
                        )
                    else:
                        self.errors.extend(plugin.errors)

        return errors

```